resources/[system]/scoreboard/server.lua

```diff
@@ -0,0 +1,101 @@
+RegisterNetEvent("scoreboard:getPlayers")
+AddEventHandler("scoreboard:getPlayers", function()
+  TriggerClientEvent("scoreboard:receivePlayers", source, players)
+end)
+
+columns = {
+  {
+    friendlyName = "Player ID",
+    defaultValue = 0,
+    position = 0
+  },
+  {
+    friendlyName = "Name",
+    defaultValue = "***Invalid***",
+    position = 1
+  }
+}
+
+players = {}
+
+RegisterNetEvent("scoreboard:requestColumns")
+AddEventHandler("scoreboard:requestColumns", function()
+  players[source] = {}
+  for id, columnData in pairs(columns) do
+    players[source][columnData.friendlyName] = columnData.defaultValue
+  end
+  players[source]["Player ID"] = source
+  players[source]["Name"] = GetPlayerName(source)
+  TriggerClientEvent("scoreboard:receiveColumns", source, columns)
+end)
+
+--[[
+Event name: scoreboard:addColumn
+Description: Adds a column to the scoreboard in the specified position with a default value.
+Parameters:
+_friendlyName: Name of the column you want to add (shows on scoreboard).
+_defaultValue: Default value of the column.
+_position: The position of the column.
+]]
+AddEventHandler("scoreboard:addColumn", function(_friendlyName, _defaultValue, _position)
+  columns[#columns+1] = {
+    friendlyName = _friendlyName,
+    defaultValue = _defaultValue or 0,
+    position = _position
+  }
+  for _, playerId in pairs(GetPlayers()) do
+    if players[tonumber(playerId)] ~= nil then
+      players[tonumber(playerId)][_friendlyName] = _defaultValue
```

Comment:
Can this not be state bag-based? Why does this have its own storage?

What if there's any other formatters?

Did you even compare to the MTA dxscoreboard API?

Reply:
We could add the columns to the GlobalState however as for the player we cannot get player bags outside the scope so this would lead into the same issue the old scoreboard had.